### PR TITLE
Task Completed

### PR DESCRIPTION
**Pull request by RohanSingh-inno**

### CODE REVIEW

**Comment by vivekpan27** [comment]
Please first check if files which have been pushed are correct or not.
I can see some files which are not supposed to be pushed.

**Comment by vivekpan27** [comment]
@RohanSingh-inno Please resolve the comment ASAP.

**Comment by RohanSingh-inno** [comment]
ok sir

File: web/modules/custom/first_custom_module/first_custom_module.info.yml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+name: first custom module
```
Comment (vivekpan27) [inline web/modules/custom/first_custom_module/first_custom_module.info.yml:1]:
```
name: First custom module
```

File: web/modules/custom/first_custom_module/first_custom_module.info.yml [hunk 2]
```diff
@@ -0,0 +1,6 @@
...
+type: module
+description: 'Custom module for Drupal ticket FT2023-339'
+package: Custom
+core_version_requirement: ^9 || ^10
```
Comment (vivekpan27) [inline web/modules/custom/first_custom_module/first_custom_module.info.yml:5]: core_version_requirement: ^10

File: web/modules/custom/first_custom_module/first_custom_module.links.menu.yml [hunk 3]
```diff
@@ -0,0 +1,19 @@
+first_custom_module.Page1:
+  title: 'Introduction'
+  description: 'This is the introduction page'
+  route_name: first_custom_module.Page1
```
Comment (vivekpan27) [inline web/modules/custom/first_custom_module/first_custom_module.links.menu.yml:4]: There are no quotes here in the RHS and in the next link you have added quotes.
Please try to follow a single approach.
Check in core files how it is done.

File: web/modules/custom/first_custom_module/first_custom_module.links.menu.yml [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+first_custom_module.Page2:
+  title: 'Introduction-2'
+  description: 'This is the second intro page'
+  route_name: 'first_custom_module.Page2' 
+  weight: 100 
+  menu_name: 'main'
+  options:
+    attributes:
+      target: '_blank'
```
Comment (vivekpan27) [inline web/modules/custom/first_custom_module/first_custom_module.links.menu.yml:19]: This red circle indicates that the new line at end of file is missing.
Did you run the code through the plugin I mentioned in slack?

File: web/modules/custom/first_custom_module/first_custom_module.permissions.yml [hunk 5]
```diff
@@ -0,0 +1,3 @@
+information permission:
+  title : Change Information
+  description: You can change the value of this
```
Comment (vivekpan27) [inline web/modules/custom/first_custom_module/first_custom_module.permissions.yml:3]: Same red circle here

File: web/modules/custom/first_custom_module/first_custom_module.routing.yml [hunk 6]
```diff
@@ -0,0 +1,15 @@
+first_custom_module.Page1:
+  path: '/Page1'
+  defaults:
+    _controller: 'Drupal\first_custom_module\Controller\BasicPageController::Page1'
```
Comment (vivekpan27) [inline web/modules/custom/first_custom_module/first_custom_module.routing.yml:4]: does class function name ever start with a capital letter?

File: web/modules/custom/first_custom_module/first_custom_module.routing.yml [hunk 7]
```diff
@@ -0,0 +1,15 @@
+first_custom_module.Page1:
```
Comment (vivekpan27) [inline web/modules/custom/first_custom_module/first_custom_module.routing.yml:1]:
```
first_custom_module.page1:
```

File: web/modules/custom/first_custom_module/first_custom_module.routing.yml [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+  path: '/Page1'
```
Comment (vivekpan27) [inline web/modules/custom/first_custom_module/first_custom_module.routing.yml:2]: Does path ever start with a capital letter?

File: web/modules/custom/first_custom_module/src/Controller/BasicPageController.php [hunk 9]
```diff
@@ -0,0 +1,43 @@
+<?php
+namespace Drupal\first_custom_module\Controller;
+
+use Drupal\Core\Controller\ControllerBase;
+
+/**
+ * Class for showing the pages.
+ */
+class BasicPageController extends ControllerBase
+{
```
Comment (vivekpan27) [inline web/modules/custom/first_custom_module/src/Controller/BasicPageController.php:10]: The curly braces should start in the prevous line or this line?

File: web/modules/custom/first_custom_module/src/Controller/BasicPageController.php [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+
+    /**
+     * Display the markup.
+     *
+     * @return array
+     *   Return markup array.
+     */
+    public function Page1()
+    {
+        $user = \Drupal::currentUser()->getAccountName();
```
Comment (vivekpan27) [inline web/modules/custom/first_custom_module/src/Controller/BasicPageController.php:20]: This kind of service calling is not acceptable in classes.
Please use dependency injection here.

File: web/modules/custom/first_custom_module/src/Controller/BasicPageController.php [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+     * @return array
+     *   Return markup array.
+     */
+    public function Page2()
+    {
+        return [
+            '#title' => 'Hi arijit this side',
+            '#data' => 'This is the data field'
+        ];
+    }
```
Comment (vivekpan27) [inline web/modules/custom/first_custom_module/src/Controller/BasicPageController.php:39]: Indentation is all wrong.

**Review by vivekpan27 — CHANGES_REQUESTED** [review]

File: web/modules/custom/first_custom_module/first_custom_module.permissions.yml (hunk 5 above)
Comment (vivekpan27) [reply web/modules/custom/first_custom_module/first_custom_module.permissions.yml]: not fixed still

File: web/modules/custom/first_custom_module/src/Controller/BasicPageController.php [hunk 12]
```diff
@@ -0,0 +1,54 @@
...
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Drupal\Core\Controller\ControllerBase;
+use Drupal\Core\Session\AccountInterface;
+
+/**
+ * Class for showing the pages.
+ */
+class BasicPageController extends ControllerBase{
+
+    protected $account;
```
Comment (vivekpan27) [inline web/modules/custom/first_custom_module/src/Controller/BasicPageController.php:14]: Function docblock missing.

File: web/modules/custom/first_custom_module/src/Controller/BasicPageController.php (hunk 11 above)
Comment (vivekpan27) [reply web/modules/custom/first_custom_module/src/Controller/BasicPageController.php]: It's still wrong.

**Review by vivekpan27 — CHANGES_REQUESTED** [review]
Did you run the code through Drupal code sniffer as mentioned yesterday?
I don't think so.
Run it on your custom module folder, make fixes, then assign the PR back for review.

**Pull request merged**